File: OctavePSSProgram/Salter.m
# The Salter class is responsible for salting the y values from the function,
# with randomized integers between -bound and bound, which is a variable chosen
# by the user.
#
# Author: Rachel Hussmann
classdef Salter

  # This section is like the body of a Java class.
  # This section is stating that the access to the method is public and the
  # class has one method.
  methods (Access = public)

    # addSalt is the method that adds the random integer value to the y
    # values of a function.
    #
    # Parameters: salt - The Salter object, yValues - The y values from the
    # function that need to be salted, bound - The upper and lower bound of the
    # random number generator (-bound to bound)
    #
    # Returns: saltedValues - The salted y values of the function
    function saltedValues = addSalt(salt, yValues, bound)

      # Iterate through the elements in the yValues list
      for i = 1 : length(yValues)

          # Generate a new random integer between -bound and bound.
          # Add it to the current y in yValues
          yValues(i) = yValues(i) + randi([-bound, bound]);

      endfor

      #Return the newly salted values
      saltedValues = yValues;

    endfunction

  endmethods

endclassdef

File: PSSPart3/src/main/java/Salter.java
import java.util.ArrayList;
import java.util.Random;

/**
 * The Salter class holds the method responsible for salting data that it is given, specifically the y values to a
 * computed function.
 * @author Rachel Hussmann
 */
public class Salter {

    /**
     * The addSalt method adds a random value to the y values to hide the original values.
     * @param yValues The ArrayList of y values calculated using the function and the x values
     * @param origin The origin of the random function (inclusive)
     * @param bound The upper bound of the random function (exclusive)
     * @return The yValues that have been salted
     */
    public ArrayList<Double> addSalt(ArrayList<Double> yValues, int origin, int bound){

        //Create the new random object
        Random random = new Random();

        //Iterate through the yValues, add the random number to the y value, and set the new y value to the same index
        //as the old one
        for(int i = 0; i < yValues.size(); i++){
            int randomNumber = random.nextInt(origin, bound);
            double numberToSalt = yValues.get(i);
            yValues.set(i, numberToSalt+randomNumber);
        }

        //Return the new yValues
        return yValues;
    }
}
